.github/workflows/single-file-build-and-test.yml

```diff
@@ -43,5 +43,5 @@ jobs:
       - name: Build and run test
         shell: cmd
         run: |
-          cl.exe /std:c++14 single-file-test.cc
+          cl.exe /std:c++14 /W4 single-file-test.cc
```

Comment:
I was thinking of adding the warnings in CMake and building and running through that. We would probably want to add `/WX` to treat the [warnings as errors](https://learn.microsoft.com/en-us/cpp/build/reference/wx-treat-linker-warnings-as-errors?view=msvc-170).